homeassistant/components/overkiz/entity.py

```diff
@@ -0,0 +1,107 @@
+"""Parent class for every Overkiz device."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from pyoverkiz.enums import OverkizAttribute, OverkizState
+from pyoverkiz.models import Device
+
+from homeassistant.components.sensor import SensorEntityDescription
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import OverkizDataUpdateCoordinator
+from .executor import OverkizExecutor
+
+
+class OverkizEntity(CoordinatorEntity):
+    """Representation of an Overkiz device entity."""
+
+    coordinator: OverkizDataUpdateCoordinator
+
+    def __init__(
+        self, device_url: str, coordinator: OverkizDataUpdateCoordinator
+    ) -> None:
+        """Initialize the device."""
+        super().__init__(coordinator)
+        self.device_url = device_url
+        self.base_device_url, *_ = self.device_url.split("#")
+        self.executor = OverkizExecutor(device_url, coordinator)
+
+        self._attr_assumed_state = not self.device.states
+        self._attr_available = self.device.available
+        self._attr_unique_id = self.device.device_url
+        self._attr_name = self.device.label
+
+    @property
+    def device(self) -> Device:
+        """Return Overkiz device linked to this entity."""
+        return self.coordinator.data[self.device_url]
+
+    @property
+    def device_info(self) -> DeviceInfo:
```

Comment:
Can we generate this once and set it as `self._attr_device_info` in `__init__`?

Reply:
Done in 9fa8250.